assets/app/scripts/directives/runDockerImage.js

```diff
@@ -0,0 +1,127 @@
+                                        Navigate,
+                                        ProjectsService,
+                                        TaskList) {
+    return {
+      restrict: 'E',
+      scope: {
+        project: '=',
+        context: '=',
+        alerts: '='
+      },
+      templateUrl: 'views/directives/run-docker-image.html',
+      link: function($scope) {
+        $scope.app = {
+          env: {},
+          labels: {}
+        };
+
+        var stripTag = $filter('stripTag');
+        var humanizeKind = $filter('humanizeKind');
+
+        // Change image names like "openshift/hello-openshift:latest" to "hello-openshift", which can be used as an app name.
+        var getName = function() {
+          // Remove everything through the last '/'.
+          var nameAndTag = _.last($scope.import.name.split('/'));
+          return stripTag(nameAndTag);
+        };
+
+        function getResources() {
+          return DockerImageGeneratorService.getResources({
+            name: $scope.app.name,
+            image: $scope.import.name,
+            tag: $scope.import.tag,
+            ports: $scope.ports,
+            volumes: $scope.volumes,
+            env: $scope.app.env,
+            labels: $scope.app.labels
+          });
+        }
+
+        $scope.findImage = function() {
+          $scope.loading = true;
+          DockerImageGeneratorService.findImage($scope.imageName, $scope.context)
+            .then(
+              // success
+              function(response) {
+                $scope.loading = false;
+                $scope.import = response;
+                var image = $scope.import.image;
+                if (image) {
+                  $scope.app.name = getName();
+                  $scope.app.labels = {
+                    app: $scope.app.name
+                  };
+                  $scope.runsAsRoot = DockerImageGeneratorService.runsAsRoot(image);
+                  $scope.ports = ApplicationGenerator.parsePorts(image);
+                  $scope.volumes = DockerImageGeneratorService.getVolumes(image);
+                }
+              },
+              // failure
+              function(response) {
+                $scope.loading = false;
+                $scope.alerts['import-image'] = {
+                  type: "error",
+                  message: "An error occurred finding the image.",
+                  details: $filter('getErrorDetails')(response)
+                };
+              });
+          };
+
+          $scope.$watch('app.name', function() {
+            _.set($scope, 'app.labels.app', $scope.app.name);
+          });
+
+          $scope.create = function() {
+            var resources = getResources();
+            var titles = {
+              started: "Deploying image " + $scope.import.imageName + " in project " + $scope.project,
+              success: "Deployed image " + $scope.import.imageName + " in project " + $scope.project,
```

Comment:
not sure we should say "deploy" here, I assume this is just going and creating the IS, DC, Service, etc? in that case it may not really be deployed yet